### DISCUSSION TITLE
Abort sync on Application warnings

### DISCUSSION THREAD

**Discussion by slok** [argoproj/argo-cd#9255] [Q&A]
Hi! :wave: 

Today we had an incident regarding this. I'll try to give some context first:

- We are using app-of-apps pattern.
- This Application (app-of-apps), has an Application identified with the name `my-app-1`.
- Accidentally during a service migration, we created another app named the same (`my-app-1`), this new app, pointed to another repo path, and had automated prune enabled. :facepalm:
- With the bad luck that ArgoCD decided to apply the last one (new one), so it applied the settings of this other path and auto-prune, so it pruned all the old app resources (including ingress, ns...).
- And this situation was one of the main causes of the incident.

Argo was correctly [warning about having 2 resources with the same identifiers](https://github.com/argoproj/argo-cd/issues/1070). But there wasn't anything we could do at that point, it was late, everything was applied, and returning 5xx.

:question: My question is: app-of-apps is a very critical Application (if not the most one) in our ArgoCD setup, is there an option or a way of avoiding syncing the app with any kind of warning?

Thanks!

**Comment by rishabh625** [comment]
May be I didn't get you, but u could have manual sync set on application does that answers?

**Reply by slok** [reply]
Hi @rishabh625, thanks for the answer!

That would not be an option, in this case, we have automatic syncing, and we avoid manual management (unless there is an exception like something failing). However, we are looking for something wherein the case of errors that can be known in advance (like a resource being declared twice)  ArgoCD abort the sync as a fail-fast mode or similar.

**Reply by rishabh625** [reply]
I think this can be requested as feature, while creating application you could pass the behaviour if duplicate resource detected set sync manual or abort. You should raise an issue , may be i'll pick to work on it.

**Reply by slok** [reply]
Done! thanks
